### PR TITLE
[EC-547] members details dialog improvements

### PR DESCRIPTION
**Pull request by coroiu**
## Type of change

```
- [ ] Bug fix
- [x] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

Modernize the user modal and prepare it for the new tab structure.

I've also moved and renamed some files for better organization and isolation.

## Screenshots

image

## Before you submit

- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

### CODE REVIEW

File: apps/web/src/app/organizations/manage/people.component.ts [hunk 1]
```diff
@@ -240,36 +241,29 @@ export class PeopleComponent
...
+        break;
+      case MemberDialogResult.Deleted:
+        this.removeUser(user);
+        break;
+      case MemberDialogResult.Revoked:
+        this.load();
+        break;
+      case MemberDialogResult.Restored:
+        this.load();
+    }
```
Comment (shane-melton) [inline apps/web/src/app/organizations/manage/people.component.ts:266]: suggestion: We can reduce some of the repeated `this.load()` calls by combining some of these cases.
```suggestion
    switch (result) {
      case MemberDialogResult.Deleted:
        this.removeUser(user);
        break;
      case MemberDialogResult.Saved:
      case MemberDialogResult.Revoked:
      case MemberDialogResult.Restored:
        this.load();
        break;
    }
```

File: apps/web/src/app/organizations/manage/user-dialog/user-dialog.component.ts [hunk 2]
```diff
@@ -15,21 +17,35 @@ import { OrganizationUserUpdateRequest } from "@bitwarden/common/models/request/
...
+  Collections = 2,
+}
+
+export interface UserDialogParams {
+  name: string;
+  organizationId: string;
+  organizationUserId: string;
+  usesKeyConnector: boolean;
+  initialTab?: UserDialogTab;
+}
```
[the remark this answers is not in the document]
Comment (shane-melton) [reply apps/web/src/app/organizations/manage/user-dialog/user-dialog.component.ts]: We might as well remove them from the dialog params to remove unnecessary noise. The dialog component is already fetching the user internally, so I see no reason to pass them. We should be able to just set them within `ngOnInit()` in the dialog component.

**Review by shane-melton — APPROVED** [review]
Nice work! I had one tiny suggestion, but no need to hold up the PR for it. 

Approving, but feel free to also make that small change to the dialog params to remove the "redundant" fields that you mentioned in another comment.

File: apps/web/src/app/organizations/manage/people.component.ts (hunk 1 above)
Comment (coroiu) [reply apps/web/src/app/organizations/manage/people.component.ts:266]: Huh, I thought I tried this and got hit with `eslint(no-fallthrough)` but apparently your suggestion works fine 🤷 
I must've missed a `break;` statement or something, thanks for pointing this out!

Comment (coroiu) [reply apps/web/src/app/organizations/manage/people.component.ts:266]: Done!

File: apps/web/src/app/organizations/manage/user-dialog/user-dialog.component.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (coroiu) [reply apps/web/src/app/organizations/manage/user-dialog/user-dialog.component.ts]: It seems it doesn't actually fetch the user, it just fetches the `OrganizationUser` which doesn't contain those fields

**Pull request merged**